File: codingame/old/onboarding.php
#!/usr/bin/php
<?php
for(;;){
	fscanf(STDIN,"%d",$n);
	$mi=9999999;
	$name='';
	for($i=0;$i<$n;$i++){
		fscanf(STDIN,"%s%f",$enemy,$dist);
		if($mi>$dist){
			$mi=$dist;
			$name=$enemy;
		}
	}
	echo $name.PHP_EOL;
}

File: codingame/old/onboarding.pl
#!/usr/bin/perl
use strict;
use warnings;
use feature 'say';
select(STDOUT); $| = 1;
for(;;){
	my $n=<>;
	my $mi=9999999;
	my $name='';
	for(my $i=0;$i<$n;$i++){
		my($enemy,$dist)=split(' ',<>);
		if($mi>$dist){
			$mi=$dist;
			$name=$enemy;
		}
	}
	say $name;
}